aurora/src/main/java/com/market/banica/aurora/config/Publishers.java

```diff
@@ -0,0 +1,115 @@
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+@EnableMBeanExport
+@ManagedResource
+@Configuration
+public class Publishers {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Publishers.class);
+
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
+
+    private final String publishersFileName;
+
+    CopyOnWriteArrayList<String> publishersList;
+
+
+    @Autowired
+    public Publishers(@Value("${aurora.channels.publishers}") String fileName) {
+        this.publishersFileName = fileName;
+        this.publishersList = this.readPublishersFromFile();
+    }
+
+    @ManagedOperation
+    public List<String> getPublishersList() {
+        return publishersList;
+    }
+
+    @ManagedOperation
+    public void addPublisher(String publisher) {
+        if (publishersList.contains(publisher)) {
+            throw new IllegalArgumentException("Publisher is already defined.");
+        }
+        publishersList.add(publisher);
+        writeBackUp();
+    }
+
+    @ManagedOperation
+    public void deletePublisher(String publisher) {
+        if (!publishersList.contains(publisher)) {
+            throw new IllegalArgumentException("Publisher is not defined.");
+        }
+        publishersList.remove(publisher);
+        writeBackUp();
+    }
+
+    protected void writeBackUp() {
+        try {
+            lock.writeLock().lock();
+            LOGGER.info("Writing back-up to json");
+
+            ObjectWriter objectWriter = new ObjectMapper().writerWithDefaultPrettyPrinter();
+
+            try (Writer output = new OutputStreamWriter(new FileOutputStream(ApplicationDirectoryUtil.getConfigFile(publishersFileName)), UTF_8)) {
+
+                String jsonData = getStringFromList(publishersList, objectWriter);
+
+                output.write(jsonData);
+
+                LOGGER.info("Back-up written successfully");
+            } catch (IOException e) {
+                LOGGER.error("Exception thrown during writing back-up");
+            }
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    private CopyOnWriteArrayList<String> readPublishersFromFile() {
+        LOGGER.debug("Reading channel property from file.");
+        try (InputStream input = new FileInputStream(ApplicationDirectoryUtil.getConfigFile(publishersFileName))) {
+
+            return new ObjectMapper().readValue(input,
+                    new TypeReference<CopyOnWriteArrayList<String>>() {
+                    });
+
+        } catch (IOException e) {
+            //log exception
+        }
+        return new CopyOnWriteArrayList<>();
+    }
+
+
+    private String getStringFromList(CopyOnWriteArrayList<String> data, ObjectWriter objectWriter)
```

Comment:
You can extract this method in separate util class and make it generic to accept Object class as intup as I saw same logic has been used on several places. You can name the method getObjectAsJsonString if you want.

Reply:
Should I name the class Utility ?

Reply:
added